src/server/server.js

```diff
@@ -302,28 +301,22 @@ Server.prototype.start = async function(done) {
         }
     }
     this.configureRoutes();
-    this._server = this.app.listen(this.opts.port, err => {
-        if (err) {
-            return done(err);
-        }
-
-        // eslint-disable-next-line no-console
-        console.log(`listening on port ${this.opts.port}`);
-
-        // Enable the websocket handling
-        this._wss = new WebSocketServer({server: this._server});
-        this._wss.on('connection', (socket, req) => {
-            socket.upgradeReq = req;
-            const client = new Client(this._logger, socket);
-            NetworkTopology.onConnect(client);
-        });
-
-        // Enable Vantage
-        if (this.opts.vantage) {
-            new Vantage(this).start(this.opts.vantagePort);
-        }
-        done();
+    this._server = this.app.listen(this.opts.port);
```

Comment:
don't we need `await` here?

Reply:
Surprisingly, no. `this.app.listen` returns an express server object even when called without a callback. Errors are thrown as normal if there is an error when trying to start the server (eg, `EADDRINUSE`).